### PR TITLE
Refactor and document `Language::Python`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Extracted from https://github.com/Homebrew/brew/pull/8339 for easier review.

### CODE REVIEW

File: Library/Homebrew/language/python.rb [hunk 1]
```diff
@@ -112,6 +113,24 @@ def detected_python_shebang(formula = self)
...
+
+      PYTHON_VIRTUALENV_URL_MOJAVE =
+        "https://files.pythonhosted.org/packages/b1/72" \
+        "/2d70c5a1de409ceb3a27ff2ec007ecdd5cc52239e7c74990e32af57affe9" \
+        "/virtualenv-15.2.0.tar.gz"
+      private_constant :PYTHON_VIRTUALENV_URL_MOJAVE
+      PYTHON_VIRTUALENV_SHA256_MOJAVE =
+        "1d7e241b431e7afce47e77f8843a276f652699d1fa4f93b9d8ce0076fd7b0b54"
+      private_constant :PYTHON_VIRTUALENV_SHA256_MOJAVE
+
```
Comment (maxim-belkin) [inline Library/Homebrew/language/python.rb:133]: 1. Is there a reason why this is not
    ```rb
    if MacOS.version
      PYTHON_VIRTUALENV_URL = ...
    ...
    else
      PYTHON_VIRTUALENV_URL = ...
    ...
    ```

2. Below we use `MacOS.version` but there is no `require "os/mac/version"` or `require "os/mac"`. Are we sure we don't need it on a Mac?

3. On Linux `MacOS.version` defaults to `NULL`, so Linux defaults to using `PYTHON_VIRTUALENV_URL_MOJAVE` -- I think it's better to make it default to using virtualenv-16.7.4.

Comment (reitermarkus) [reply Library/Homebrew/language/python.rb:133]: I'm not really sure how this should behave.

Comment (Bo98) [reply Library/Homebrew/language/python.rb:133]: `PYTHON_VIRTUALENV_URL_MOJAVE` purely exists for system Python 2, which I don't think we really support anymore.

Comment (MikeMcQuaid) [reply Library/Homebrew/language/python.rb:133]: Can try to kill it here.

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/language/python.rb [hunk 2]
```diff
@@ -115,13 +116,9 @@ module Virtualenv
...
-            if MacOS.version > :mojave
-              url PYTHON_VIRTUALENV_URL
-              sha256 PYTHON_VIRTUALENV_SHA256
-            else
-              url PYTHON_VIRTUALENV_URL_MOJAVE
-              sha256 PYTHON_VIRTUALENV_SHA256_MOJAVE
-            end
+            url "https://files.pythonhosted.org/packages/11/74" \
+                "/2c151a13ef41ab9fb43b3c4ff9e788e0496ed7923b2078d42cab30622bdf/virtualenv-16.7.4.tar.gz"
+            sha256 "94a6898293d07f84a98add34c4df900f8ec64a570292279f6d91c781d37fd305"
```
Comment (maxim-belkin) [inline Library/Homebrew/language/python.rb:121]: Probably not in this PR, but we need to consider using a newer version of `virtualenv`. 
Versions prior to `v20.0.0` belong to the "legacy" branch (the latest one being 16.7.10 -- maybe we could use it here even now) and the latest stable is: v20.0.31

https://virtualenv.pypa.io/en/latest/changelog.html

Comment (Bo98) [reply Library/Homebrew/language/python.rb:121]: Using v20 is complicated as it requires a bunch of dependencies.

16.7.10 should be trivial to update to however.

**Pull request merged**